### PR TITLE
Fix MacOS building (master branch)

### LINKED ISSUE #122: can't build engine in mac os x
mac version 10.14.3

error info:
```
# robert @ Paul in /opt/openssl-1.1.1/engine/build on git:master o [14:01:21]
$ sudo cmake --build . --target install --config Release
Scanning dependencies of target test_tlstree
[  2%] Building C object CMakeFiles/test_tlstree.dir/test_tlstree.c.o
[  4%] Linking C executable bin/test_tlstree
[  4%] Built target test_tlstree
Scanning dependencies of target gost_core
[  6%] Building C object CMakeFiles/gost_core.dir/gost89.c.o
[  8%] Building C object CMakeFiles/gost_core.dir/gosthash.c.o
[ 10%] Building C object CMakeFiles/gost_core.dir/gosthash2012.c.o
[ 12%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_core.c.o
[ 14%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_defines.c.o
[ 16%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_galois_precompiled.c.o
[ 18%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_precompiled.c.o
[ 20%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_cipher.c.o
[ 22%] Building C object CMakeFiles/gost_core.dir/gost_ec_keyx.c.o
[ 24%] Building C object CMakeFiles/gost_core.dir/gost_ec_sign.c.o
[ 26%] Building C object CMakeFiles/gost_core.dir/gost_omac.c.o
[ 28%] Building C object CMakeFiles/gost_core.dir/gost_omac_acpkm.c.o
[ 30%] Linking C static library libgost_core.a
[ 30%] Built target gost_core
Scanning dependencies of target gost12sum
[ 32%] Building C object CMakeFiles/gost12sum.dir/gost12sum.c.o
[ 34%] Linking C executable bin/gost12sum
[ 34%] Built target gost12sum
Scanning dependencies of target gost_engine
[ 36%] Building C object CMakeFiles/gost_engine.dir/e_gost_err.c.o
[ 38%] Building C object CMakeFiles/gost_engine.dir/gost_asn1.c.o
[ 40%] Building C object CMakeFiles/gost_engine.dir/gost_crypt.c.o
[ 42%] Building C object CMakeFiles/gost_engine.dir/gost_ctl.c.o
[ 44%] Building C object CMakeFiles/gost_engine.dir/gost_eng.c.o
[ 46%] Building C object CMakeFiles/gost_engine.dir/gost_keywrap.c.o
[ 48%] Building C object CMakeFiles/gost_engine.dir/gost_params.c.o
[ 50%] Building C object CMakeFiles/gost_engine.dir/gost_keyexpimp.c.o
[ 52%] Building C object CMakeFiles/gost_engine.dir/gost_ameth.c.o
[ 54%] Building C object CMakeFiles/gost_engine.dir/gost_md.c.o
[ 56%] Building C object CMakeFiles/gost_engine.dir/gost_md2012.c.o
[ 58%] Building C object CMakeFiles/gost_engine.dir/gost_pmeth.c.o
[ 60%] Building C object CMakeFiles/gost_engine.dir/gost_omac.c.o
[ 62%] Building C object CMakeFiles/gost_engine.dir/gost_omac_acpkm.c.o
[ 64%] Linking C shared module bin/gost.so
[ 64%] Built target gost_engine
Scanning dependencies of target gostsum
[ 66%] Building C object CMakeFiles/gostsum.dir/gostsum.c.o
[ 68%] Linking C executable bin/gostsum
[ 68%] Built target gostsum
Scanning dependencies of target test_params
[ 70%] Building C object CMakeFiles/test_params.dir/test_params.c.o
[ 72%] Linking C executable bin/test_params
ld: can't link with bundle (MH_BUNDLE) only dylibs (MH_DYLIB) file 'bin/gost.so' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/test_params] Error 1
make[1]: *** [CMakeFiles/test_params.dir/all] Error 2
make: *** [all] Error 2
```
can you help me?

### PR DESCRIPTION
**Pull request by DJm00n**
Fixes #122

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ find_package(OpenSSL 1.1.1 REQUIRED)
 include_directories(${OPENSSL_INCLUDE_DIR})
 
 if (CMAKE_C_COMPILER_ID MATCHES "Clang")
- add_compile_options(-O2 -Werror -Wall -Wno-unused-parameter -Wno-unused-function -Wno-missing-braces -ggdb -Qunused-arguments)
```
[the remark this answers is not in the document]
Comment (DJm00n) [reply CMakeLists.txt]: > This branch is 34 commits ahead, 245 commits behind master.

It looks that `openssl_1_1_0` branch has outdated `CMakeLists.txt`. Do you plan to integrate changes from `main` to it?
Or maybe we can properly fix `main` branch to build with `1.1.1` OpenSSL?

[the remark this answers is not in the document]
Comment (beldmit) [reply CMakeLists.txt]: Branch 1_1_0 is compatible with OpenSSL 1.1.1 and OpenSSL 1.1.1 does not have any advantages related to GOST comparing to 1.1.0.

OpenSSL master has a differense in the constification of some functions, so it causes compilation errors. Using master (_unstable_ branch) for working with stable OpenSSL versions seems unreasonable.

Could you backport to 1_1_0 only MAC-related fixes?

**Comment by chipitsine** [comment]
вроде немного осталось ?

и добавим `osx` в travis-ci

**Comment by DJm00n** [comment]
@chipitsine убрал правку флагов сборки отсюда и сдела второй пул реквест в ветку 1_1_0

**Pull request merged**